test/lit/passes/merge-blocks-eh.wast

```diff
@@ -0,0 +1,214 @@
+          (call $import)
+          (unreachable)
+        )
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block-try_catch_all_i32 (type $0)
+  ;; CHECK-NEXT:  (tuple.drop 2
+  ;; CHECK-NEXT:   (block $catch (type $1) (result i32 exnref)
+  ;; CHECK-NEXT:    (try_table (catch_ref $i32 $catch)
+  ;; CHECK-NEXT:     (call $import)
+  ;; CHECK-NEXT:     (unreachable)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_all_i32
+    ;; Send a tag value + exnref. We don't handle this yet TODO
+    (tuple.drop 2
+      (block $catch (result i32 exnref)
+        (try_table (catch_ref $i32 $catch)
+          (call $import)
+          (unreachable)
+        )
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block-try_catch_multi_partial (type $0)
+  ;; CHECK-NEXT:  (tuple.drop 2
+  ;; CHECK-NEXT:   (block $outer (type $1) (result i32 exnref)
+  ;; CHECK-NEXT:    (block $inner
+  ;; CHECK-NEXT:     (try_table (catch_ref $i32 $outer) (catch_all $inner)
+  ;; CHECK-NEXT:      (call $import)
+  ;; CHECK-NEXT:      (unreachable)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_multi_partial
+    ;; Two possible tags can be thrown and caught, and we can optimize one.
+    (tuple.drop 2
+      (block $outer (result i32 exnref)
+        (drop
+          (block $inner (result exnref)
+            (try_table (catch_ref $i32 $outer) (catch_all_ref $inner)
+              (call $import)
+              (unreachable)
+            )
+          )
+        )
+        (unreachable)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block_try_catch_multi-fail (type $0)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (block $catch (result exnref)
+  ;; CHECK-NEXT:    (try_table (catch $exnref $catch) (catch_all_ref $catch)
+  ;; CHECK-NEXT:     (call $import)
+  ;; CHECK-NEXT:     (unreachable)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block_try_catch_multi-fail
+    (drop
+      ;; This block is sent an exnref in two ways: once as the contents of a tag
+      ;; and once as the ref of a catch_all_ref. We can remove the latter but
+      ;; not the former, and they go to the same block, so we cannot optimize.
+      (block $catch (result exnref)
+        (try_table (catch $exnref $catch) (catch_all_ref $catch)
+          (call $import)
+          (unreachable)
+        )
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block-try_catch_all (type $0)
+  ;; CHECK-NEXT:  (block $catch
+  ;; CHECK-NEXT:   (try_table (catch_all $catch)
+  ;; CHECK-NEXT:    (call $import)
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_all
+    ;; Without _ref, there is nothing to optimize (and we should not error).
```

Comment:
But wouldn't this not be optimized anyway because there's no `drop`? (so not necessarily because there's no `_ref`)

Reply:
Good point, I updated the comment. Yes, since we send no ref, there is no drop here either, so there is definitely nothing to optimize. This really just checks we don't error somehow.